nominal/core.py

```diff
@@ -751,6 +941,96 @@ def _search_datasets(self) -> Iterable[Dataset]:
         response = self._catalog_client.search_datasets(self._auth_header, request)
         for ds in response.results:
             yield Dataset._from_conjure(self, ds)
+        
+    def _get_user_rid_from_email(self, user_email: str) -> str:
+        request = authentication_api.SearchUsersRequest(
+            query=authentication_api.SearchUsersQuery(
+                exact_match=user_email,
+            )
+        )
+        response = self._authentication_client.search_users_v2(
+            self._auth_header,
+            request
+        )
+        if len(response.results) == 0:
+            raise ValueError(f"User {user_email} not found")
+        if len(response.results) > 1:
+            raise ValueError(f"Found multiple users with email {user_email}")
+        return response.results[0].rid
+
+    def create_checklist(
+        self,
+        assignee_email: str,
+        title: str,
+        checks: Sequence[CreateCheck],
+        checklist_variables: Sequence[CreateChecklistVariable],
+        default_ref_name: str | None = None,
+        commit_message: str | None = None,
+        description: str | None = None,
+        properties: Mapping[str, str] | None = None,
+        labels: Sequence[str] = (),
+        is_published: bool = True,
+    ) -> Checklist:
+        
+        conjure_checklist_variables = _batch_create_checklist_variable_to_conjure(
+            checklist_variables, self._auth_header, self._compute_representation_client, default_ref_name
+        )
+
+        conjure_checks = _batch_create_check_to_conjure(
+            checks, self._auth_header, self._compute_representation_client, default_ref_name
+        )
+    
+        request = scout_checks_api.CreateChecklistRequest(
+            commit_message=commit_message if commit_message is not None else "",
+            assignee_rid=self._get_user_rid_from_email(assignee_email),
+            title=title,
+            description=description,
+            # TODO(ritwikdixit): support functions
+            functions={},
+            properties={} if properties is None else dict(properties),
+            labels=list(labels),
+            checks=conjure_checks,
+            checklist_variables=conjure_checklist_variables,
+            is_published=is_published,
+        )
+
+        response = self._checklist_api_client.create(self._auth_header, request)
+        # TODO(ritwikdixit): remove the need for an extraneous network request
```

Comment:
Not ideal for now - we technically send a request to the representation service to resolve the expression representations of the checklist variables and the checks, when we had them from the start.. but it's not super easy to match them up between the pre / post request checks, variables